app/models/miq_server.rb

```diff
@@ -369,6 +369,13 @@ def monitor_loop
       _log.info "Server Monitoring Complete - Timings: #{timings.inspect}" unless timings[:total_time] < server_log_timings_threshold
       sleep monitor_poll
     end
+  rescue Interrupt => e
+    _log.info("Received #{e.message} signal, killing server")
+    self.class.kill
+    exit 1
```

Comment:
@Fryguy I agree, the alternative was to create a separate method (like `#shutdown_and_exit`) and just call it from here.

I thought that was a bit over*kill*, but I have no real opinion.